### PR TITLE
kernel/generic: add kernelTests automatically

### PR DESCRIPTION
**Pull request by Atemu**
###### Motivation for this change

Tests for different kernels had to be added manually and many didn't have that done. Now, they're simply generated automatically. 

I didn't build all tests again but the couple I did run were the same drv as before.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Atemu** [comment]
Fixed a rebasing error.

**Comment by Atemu** [comment]
Fixed merge conflict. Could you review this @NeQuissimus?

File: pkgs/os-specific/linux/kernel/generic.nix [hunk 1]
```diff
@@ -184,7 +185,16 @@ let
...
     passthru = kernel.passthru // (removeAttrs passthru [ "passthru" ]);
-    tests = kernelTests;
+    tests = let
+      overridableKernel = finalKernel // {
+        override = args:
+          lib.warn (
+            "override is stubbed for NixOS kernel tests, not applying changes these arguments: "
+            + toString (lib.attrNames (if lib.isAttrs args then args else args {}))
+          ) overridableKernel;
+      };
```
Comment (Atemu) [inline pkgs/os-specific/linux/kernel/generic.nix:195]: For some strange reason, the kernel in here does not have .override or .overrideDerivation, only .overrideAttrs.

**Comment by Atemu** [comment]
Rebased.

**Comment by kevincox** [comment]
Sorry for the delay, this slipped through the cracks. Unfortunately I am really not familiar with this part of NixOS and don't feel comfortable reviewing. I suggest trying to find someone who has worked on this code recently if possible.

**Comment by Atemu** [comment]
I think Marvin broke @timokau. It should've gone and looked for a new reviewer but only added `awaiting_reviewer` back.

**Comment by Atemu** [comment]
Rebased and removed the manually declared tests from all kernels (missed some initially)

**Comment by timokau** [comment]
> I think Marvin broke @timokau. It should've gone and looked for a new reviewer but only added `awaiting_reviewer` back.

Sorry for the late response. I think the bot might have re-requested someone with a pending review request. That doesn't show up in the GitHub UI if I recall correctly (https://github.com/timokau/marvin-mk2/issues/86).

It seems like the bot is not really sustainable with me as the sole maintainer given my current priorities. Maybe it would be better to shut it down. It would be sad to abort this experiment without exploring it a bit further, but that may still be better than the current state. The bot is too spammy in its current configuration and may actually put off contributors or generate unnecessary stress. I think I will post a comment to https://github.com/timokau/marvin-mk2/issues/34 soon-ish and ask for peoples opinions.

We should probably continue this discussion there, since it's a bit off topic here. Unfortunately I don't have the spare capacity to review this right now.

**Review by NeQuissimus — APPROVED** [review]

**Comment by kevincox** [comment]
@Atemu looks like some merge conflicts. If you resolve them I'll merge.

**Comment by kevincox** [comment]
Thanks! and sorry for the poorly managed review process.

**Pull request merged**